### PR TITLE
recipes/zeroing out gradients.py 번역 (#120)

### PR DESCRIPTION
**Pull request by codingbowoo**
## 라이선스 동의
_변경해주시는 내용에 BSD 3항 라이선스가 적용됨을 동의해주셔야 합니다._<br />
_더 자세한 내용은 [기여하기 문서](https://github.com/9bow/PyTorch-tutorials-kr/blob/master/CONTRIBUTING.md)를 참고해주세요._<br />
_동의하시면 아래 `[ ]`를 `[x]`로 만들어주세요._<br />

- [x] 기여하기 문서를 확인하였으며, 본 PR 내용에 BSD 3항 라이선스가 적용됨에 동의합니다.

## 관련 이슈 번호
- **이슈 번호**: #120 

## PR 종류
_이 PR에 해당되는 종류 앞의 `[ ]`을 `[x]`로 변경해주세요._<br />
- [ ] 오탈자를 수정하거나 번역을 개선하는 기여
- [x] 번역되지 않은 튜토리얼을 번역하는 기여
- [ ] 공식 튜토리얼 내용을 반영하는 기여
- [ ] 위 종류에 포함되지 않는 기여

## PR 설명
레시피 번역

### CODE REVIEW

File: recipes_source/recipes/zeroing_out_gradients.py [hunk 1]
```diff
@@ -140,36 +130,36 @@ def forward(self, x):
...
 
 ######################################################################
-# 5. Zero the gradients while training the network
+# 5. 신경망을 학습시키는 동안 변화도를 0으로 만들기
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 # 
-# This is when things start to get interesting. We simply have to loop
-# over our data iterator, and feed the inputs to the network and optimize.
+# 이제부터 흥미로운 부분을 살펴보려고 합니다.
+# 우리가 할 일은 데이터 이터레이터를 순회하면서, 신경망에 입력을 주고
```
Comment (potatochips178) [inline recipes_source/recipes/zeroing_out_gradients.py:137]: '우리가'라는 말을 쓰지 않고 '해야할 일은'으로 번역하는 건 어떨가요? 다른 부분에서는 we를 딱히 번역하지 않았는데 여기서만 번역해서 전체적으로 비슷하게 맞춰 주려면 없어도 좋을 것 같습니다.

File: recipes_source/recipes/zeroing_out_gradients.py [hunk 2]
```diff
@@ -140,36 +130,36 @@ def forward(self, x):
...
+# 신경망을 학습시킬 때 불필요한 정보를 추적하지 않도록 하기 위함입니다. 
 # 
 
-for epoch in range(2):  # loop over the dataset multiple times
+for epoch in range(2):  # 전체 데이터셋을 여러번 반복하기
 
     running_loss = 0.0
     for i, data in enumerate(trainloader, 0):
-        # get the inputs; data is a list of [inputs, labels]
+        # 입력 받기; 데이터는 [inputs, labels] 형태의 리스트다.
```
Comment (potatochips178) [inline recipes_source/recipes/zeroing_out_gradients.py:148]: '데이터는 [inputs, labels] 형태의 리스트' 라고 번역하는 건 어떨까요? 다른 주석은 명사형 어미로 끝나는데 여기만 종결어미로 끝나니까 살짝 어색해 보여서요.

**Comment by JaeLee18** [comment]
컨트리뷰톤 리뷰 진행하겠습니다.

File: recipes_source/recipes/zeroing_out_gradients.py (hunk 1 above)
Comment (codingbowoo) [reply recipes_source/recipes/zeroing_out_gradients.py:137]: 오 생각해보지 못한 부분이에요 감사합니다 :) ```여기서 할 일은``` 으로 수정하려고 합니다!

File: recipes_source/recipes/zeroing_out_gradients.py (hunk 2 above)
Comment (codingbowoo) [reply recipes_source/recipes/zeroing_out_gradients.py:148]: 좋아요!

**Review by JaeLee18 — CHANGES_REQUESTED** [review]

File: recipes_source/recipes/zeroing_out_gradients.py [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
-
-There are cases where it may be necessary to zero-out the gradients of a
-tensor. For example: when you start your training loop, you should zero
-out the gradients so that you can perform this tracking correctly.
-In this recipe, we will learn how to zero out gradients using the
-PyTorch library. We will demonstrate how to do this by training a neural
-network on the ``CIFAR10`` dataset built into PyTorch.
-
-Setup
+신경망을 학습시킬 때, 경사 하강법을 거쳐 모델 정확도를 높일 수 있습니다. 경사 하강법은 간단히
```
[the remark this answers is not in the document]
Comment (codingbowoo) [reply recipes_source/recipes/zeroing_out_gradients.py:9]: ```요컨대``` 번역할 때 구어체 느낌이 강해서 ```간단히 설명해```로 풀어 설명한 바 있습니다. **In short**는 역시 요컨대로 번역하는 것이 자연스러울까요?

(현재) 경사 하강법은 간단히 설명해 모델의 가중치와 편향을 약간씩 수정하면서 손실(또는 오류)를 최소화하는 과정입니다.
(수정) 요컨대, 경사 하강법은 모델의 가중치와 편향을 약간씩 수정하면서 손실(또는 오류)를 최소화하는 과정입니다.

File: recipes_source/recipes/zeroing_out_gradients.py [hunk 4]
```diff
@@ -1,37 +1,29 @@
...
-PyTorch library. We will demonstrate how to do this by training a neural
-network on the ``CIFAR10`` dataset built into PyTorch.
-
-Setup
+신경망을 학습시킬 때, 경사 하강법을 거쳐 모델 정확도를 높일 수 있습니다. 경사 하강법은 간단히 
+설명해 모델의 가중치와 편향을 약간씩 수정하면서 손실(또는 오류)를 최소화하는 과정입니다. 
+
+``torch.Tensor`` 는 PyTorch 의 핵심이 되는 클래스 입니다. 텐서를 생성할 때 
+``.requires_grad`` 속성을 ``True`` 로 설정하면, 텐서에 가해진 모든 연산을 추적합니다. 
+뒤따르는 모든 역전파 단계에서도 마찬가지입니다. 이 텐서의 변화도는 ``.grad`` 속성에 누적됩니다.
```
[the remark this answers is not in the document]
Comment (codingbowoo) [reply recipes_source/recipes/zeroing_out_gradients.py:14]: 앞으로 일어날 역전파 단계니까 제안해 주신 ```누적될 것입니다```가 좋을 것 같네요. 아래와 같이 수정하겠습니다.

뒤따르는 모든 역전파 단계에서도 마찬가지로 이 텐서의 변화도는 ``.grad`` 속성에 누적될 것입니다.

File: recipes_source/recipes/zeroing_out_gradients.py [hunk 5]
```diff
@@ -1,37 +1,29 @@
...
+모든 변화도의 축적 또는 합은 손실 텐서에서 ``.backward()``를 호출할 때 계산됩니다.
+
+텐서의 변화도를 0으로 만들어 주어야 하는 경우도 있습니다. 예를 들어 학습 과정 전체를 반복하기
```
[the remark this answers is not in the document]
Comment (codingbowoo) [reply recipes_source/recipes/zeroing_out_gradients.py:17]: 이부분 고민을 했었는데, 간결하고 쉽게 전달되는 것 같습니다. 감사합니다.

**Review by codingbowoo — COMMENTED** [review]
@JaeLee18 @potatochips178 리뷰해주신 것 대부분 반영했습니다. 감사합니다. 👍

File: recipes_source/recipes/zeroing_out_gradients.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (creduo) [reply recipes_source/recipes/zeroing_out_gradients.py:9]: 앞에서 경사하강법의 설명이 진지하게 붙었고 그를 요약한다면 요컨대가 맞을 것 같지만, 여기서는 요컨대 보다는 간단히 설명하여 등의 표현이 좀더 자연스러운것 같습니다.

**Review by creduo — APPROVED** [review]
@codingbowoo 님 번역에 수고하셨습니다. @potatochips178 님, @JaeLee18 님 리뷰 감사드립니다.
